Add package_data to setup.py for hdaccount wordlists

kclowes opened the pull request:
## What was wrong?
Web3py couldn't use the new HD Account methods because the wordlists (raw txt files) weren't getting included in the release.

Fixes #102 


## How was it fixed?
Added a line to setup.py to include them, and tested in web3 using a local install of eth-account.

#### Cute Animal Picture

![image](https://user-images.githubusercontent.com/6540608/80012492-a2ce3e00-848a-11ea-9911-9bbc0ebaa490.png)

pipermerriam commented:
I *think* CI might have caught this were we to have `usedevelop=False` in the `tox.ini` config.

https://github.com/ethereum/eth-account/blob/master/tox.ini#L24

I'm not 100% sure about this but I do believe this is something that can and should be caught by CI.

kclowes commented:
Yeah, agreed. It would have been nice to catch this in CI. I'm getting a bunch of `ImportMismatchError`s when I remove `usedevelop=true` and my google-fu is failing me. Do you know a way around that?

kclowes commented:
Ha, never mind. Fixed it! Let me know if there is a better way to fix it!

pipermerriam commented on setup.py:
```diff
@@ -47,6 +47,18 @@
     include_package_data=True,
+    package_data={"eth_account": ["hdaccount/wordlist/*.txt"]},
+    data_files=[('eth_account', [
+        'eth_account/hdaccount/wordlist/chinese_simplified.txt',
```
I think you can do this via the `MANIFEST.in` file which allows for a `*` style inclusion that will not suffer from us forgetting to update this list if we add a new word-list.

https://github.com/ethpm/ethpm-cli/blob/432e3aa40faeda6eea54530bdf3d2c81fd80d5a1/MANIFEST.in#L41

kclowes replied:
We have a `Manifest.in`, but the doctests were still failing. I _think_ we only need this for the doctests, in which case we only need the english one, so I could just include that one?

kclowes replied:
@carver @pipermerriam So turns out if I use the generated tarball file to install `eth-account`, I need the `data_files` here, but if I use the wheel to install, I don't need data_files. How concerned are we about the tarball distribution? I assume both should work?

carver replied:
Hm, this suggests that the wheel is better, all else equal: https://stackoverflow.com/a/42226284/8412986

Based on the docs though, I can't see why we would need `data_files` *and* `package_data`. So you're saying a wheel install will fail when this `data_files` entry is removed?

kclowes replied:
Yeah, it's entirely possible I'm doing something wrong, but when I do 
```
make dist # in my eth-account dir, 
cd path/to/web3 
pip install /path/to/eth-account/eth_account/dist/eth-account-0.5.0.tar.gz
```
and open up ipython to test, I get the error where it doesn't load any of the languages from eth-account's `Mnemonic.list_languages` (code is found [here](https://github.com/ethereum/eth-account/blob/master/eth_account/hdaccount/mnemonic.py#L85-L86)). The `make dist` output shows that the files get added. My one thought was that maybe there is something going on with relative/absolute paths in the `list_languages` method?

kclowes replied:
Got it! The syntax was wrong in the `MANIFEST.in` file

kclowes commented:
That makes sense, but that was the only way I could get around the `ImportMismatchError` after I changed `usedevelop` to `false`. Do you know another way to get around that error?

kclowes commented:
I also haven't figured out why quite yet, but setting `usedevelop=false` doesn't fail any tests if the `wordlist/*.txt` files aren't packaged up, even though from everything I've read, it seems like it should. The doctests do fail as expected without the wordlist files, however. Maybe there is something special about what gets installed for tests in the `tests` dir?

carver commented:
Yeah, so the way I've usually been testing this kind of thing is a test where it specifically builds a wheel and then only uses that for running the test. It's a little ugly, but it's a way to know that exactly the installed version is being tested. Like:

https://github.com/ethereum/trinity/blob/27bbf61209d81b894292e36904a8b634fcb0094a/tox.ini#L101-L105

carver commented:
Notice the `skip_install` in addition to not using develop:
https://github.com/ethereum/trinity/blob/27bbf61209d81b894292e36904a8b634fcb0094a/tox.ini#L111-L112

carver commented on MANIFEST.in:
```diff
@@ -5,7 +5,7 @@ include requirements-docs.txt
 
 # For mnemonics
-recursive-include eth_account/hdaccount/wordlist/*
+recursive-include eth_account hdaccount wordlist *.txt
```
Aha good find! By my reading of [this](https://packaging.python.org/guides/using-manifest-in/):

> recursive-include dir-pattern pat1 pat2 ... | Include all files under directories matching dir-pattern that match any of the listed patterns

I would expect that the intended approach is:
```suggestion
recursive-include eth_account/hdaccount/wordlist *.txt
```

Does that work?

carver replied:
Also, it's worth making sure you have the latest tooling with a `pip install -U pip setuptools wheel`. They have seemed pretty active.

carver commented:
It bothers me that CI was still green before the manifest syntax. Would it be possible to add the wheel-building test in, show that it fails, then push the last manifest syntax fix to show that turns it green?

kclowes commented:
@carver I think this is ready for a last look. I added the wheel-cli tests to circleci. Let me know if I should test for anything else there. Since I removed the `usedevelop=True`, the mnemonics core tests are failing in the way that I'd expect, so I didn't add that to the wheel-cli tests. Since eth-account doesn't really have cli tests like trinity, I made the tests more closely resemble the web3 tests where I just made sure that eth-account could be imported after install. I did check locally that it worked, but can push up a test if you're interested in seeing it go red.
